File: task_equivalents/llm_translations/gpt/base/r/statistical_modelling_analysis/01_linear_regression_inference.R
library(readr)

Auto <- read_csv("data/csv/Auto.csv", na = "?") %>%
na.omit()

n <- nrow(Auto)
X <- matrix(Auto$horsepower, ncol = 1)
Y <- matrix(Auto$mpg, ncol = 1)

model1_fit <- lm(Y ~ X[, 1])
summary(model1_fit)

xnew <- data.frame(X[, 1] = 98)
predict(model1_fit, newdata = xnew, interval = "confidence", level = 0.95)

File: task_equivalents/llm_translations/gpt/base/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.formula.api as smf

auto = pd.read_csv("data/csv/Auto.csv", header=0, na_values="?")
auto = auto.dropna()

model1 = smf.ols("mpg ~ horsepower", data=auto).fit()
print(model1.summary())

new_data = pd.DataFrame({"horsepower": [98]})

confidence_pred = model1.get_prediction(new_data).summary_frame(alpha=0.05)
print(confidence_pred[["mean", "mean_ci_lower", "mean_ci_upper"]])

prediction_pred = model1.get_prediction(new_data).summary_frame(alpha=0.05)
print(prediction_pred[["mean", "obs_ci_lower", "obs_ci_upper"]])